psycopg/psycopg/types/geometry.py

```diff
@@ -0,0 +1,63 @@
+"""
+Adapters for PostGIS geometries
+"""
+
+from typing import TYPE_CHECKING
+
+from psycopg.adapt import Dumper, Loader
+from psycopg.pq import Format
+from psycopg.types import TypeInfo
+
+if TYPE_CHECKING:
+    from psycopg.connection import Connection
+    from psycopg.rows import Row
+
+
+def register_shapely_adapters(conn: Connection[Row]) -> None:
+    """Register Shapely dumper and loaders.
+
+    After invoking this function on a connection, the queries retrieving
+    PostGIS geometry objects will return Shapely's shape object instances
+    in text and binary mode.
+
+    Similarly, shape onjects can be sent to the database.
+
+    :param conn: the connection on which to set up the loader and dumper
+
+    """
+    import shapely.wkb
+    from shapely.geometry.base import BaseGeometry
+
+    class GeometryBinaryLoader(Loader):
```

Comment:
You should define these functions top level in the module (you can make `shapely` a global var so that you can lazy import it and still use it  from these classes.